### PR TITLE
feat: add buyer-vendor messaging plugin

### PR DESCRIPTION
**Pull request by kjedras**
## Summary

- **Buyer-vendor messaging** — customers and sellers can exchange messages through store/vendor API routes with real-time SSE delivery
- **Content filtering** — built-in regex rules (email, phone, URL) plus admin-managed exact/contains filters to prevent contact info sharing
- **Chat blocking** — admins can block/unblock customers with reason, enforced across all endpoints
- **Admin oversight** — conversation search with name/date/context filters, blocked message log, and filter rule CRUD
- **Rate limiting** — 20 messages/minute and 5 new conversations/hour per sender via Redis sliding window
- **GDPR compliance** — anonymize workflows for buyer/vendor deletion with compensation steps and batched processing
- **Full UI** — admin and vendor dashboard pages for conversations, message detail, filters, and blocked messages
- **95 files** in the registry block with documentation (DOCUMENTATION.md, IMPLEMENTATION.md), unit tests, and integration tests

## Test plan

- [x] 88/88 manual endpoint tests pass (auth, CRUD, unread tracking, context linking, validation, ownership, content filtering, filter CRUD, chat blocking/unblocking, admin oversight, pagination, rate limiting, SSE, edge cases)
- [x] Unit tests pass: rate limiting, content filter evaluation, cursor encoding, unread counters, chat blocking
- [x] Registry.json file list matches actual files on disk (95 files)
- [x] Documentation accurate and up to date with implementation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/registry/src/messaging/api/admin/messages/events/route.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "@medusajs/framework"
+
+import { MESSAGING_REDIS_MODULE } from "../../../../modules/messaging-redis"
+import type MessagingRedisModuleService from "../../../../modules/messaging-redis/service"
+
+export const GET = async (
```
Comment (jakub-borek) [inline packages/registry/src/messaging/api/admin/messages/events/route.ts:9]: SSE is fine for simple plugin, but I don't see this scaling well for a production-ready marketplace solution, due to SSE limitations, e.g.:
- most browsers only support around 6 SSE connections, so if some users open few windows each with different chat, it will break
- it's one way communication server -> customer, not a real-time chat

I'd consider switching to full websocket implementation using Socket.io. Unless for larger marketplaces we'll use custom integrations, like TalkJS/Sendbird?

Something to think about, in the end it's better to have something than nothing

Comment (criskell) [reply packages/registry/src/messaging/api/admin/messages/events/route.ts:9]: Note: Medusa does not expose the internal HTTP server, which is necessary to initialize socket.io; a separate one must be created.
